### PR TITLE
Bug: K8s CF keypairs should be conditionally created

### PR DESCRIPTION
**Pull request by GrooveStomp**
https://app.zenhub.com/workspaces/commit-zero-5da8decc7046a60001c6db44/issues/commitdev/zero/295

- Create new var: cf_signing_enabled
- Conditionally lookup secret based on cf_signing_enabled
- Set count of k8s secret resources based on cf_sigining_enabled

### CODE REVIEW

**Comment by GrooveStomp** [comment]
How is this approach? I have yet to test it.

File: templates/kubernetes/terraform/environments/dev/main.tf [hunk 1]
```diff
@@ -19,6 +19,8 @@ module "kubernetes" {
 
   project = "<% .Name %>"
 
+  cf_signing_enabled = "<% index .Params `fileUploads` %>"
```
Comment (davidcheung) [inline templates/kubernetes/terraform/environments/dev/main.tf:22]: i think the params stores it as a string so you may need some type casting -> bool

**Comment by bmonkman** [comment]
Woo! The tests worked! 🎉

File: templates/kubernetes/terraform/environments/dev/main.tf (hunk 1 above)
Comment (bmonkman) [reply templates/kubernetes/terraform/environments/dev/main.tf:22]: Yeah, this caused the tests to fail because "yes" and "no" are not valid bool values.

**Comment by davidcheung** [comment]
another approach would be make a folder such as 
`kubernetes/terraform/modules/kubernetes/fileupload/main.tf` and have the current content of `backend-services.tf`
then in backend services 
have 
```
module "fileupload" {
  count = var.cf_signing_enabled ? 1 : 0
  source                     = "./fileupload"
  ...  some variables
}
```

File: templates/kubernetes/terraform/environments/dev/main.tf (hunk 1 above)
Comment (GrooveStomp) [reply templates/kubernetes/terraform/environments/dev/main.tf:22]: > another approach would be make a folder such as
> `kubernetes/terraform/modules/kubernetes/fileupload/main.tf` and have the current content of `backend-services.tf`
> 
> then in `kubernetes/terraform/modules/kubernetes/main.tf` have
> 
> ```
> module "fileupload" {
>   count = var.cf_signing_enabled ? 1 : 0
>   source = "./fileupload"
>   ...  some variables
> }
> ```

Oooh, I like this.

**Comment by GrooveStomp** [comment]
> another approach would be make a folder such as
> `kubernetes/terraform/modules/kubernetes/fileupload/main.tf` and have the current content of `backend-services.tf`
> 
> then in `kubernetes/terraform/modules/kubernetes/main.tf` have
> 
> ```
> module "fileupload" {
>   count = var.cf_signing_enabled ? 1 : 0
>   source = "./fileupload"
>   ...  some variables
> }
> ```

Ooh, I like this.

File: templates/kubernetes/terraform/environments/dev/main.tf (hunk 1 above)
Comment (GrooveStomp) [reply templates/kubernetes/terraform/environments/dev/main.tf:22]: Ha ha, easy oversight.  I would have looked at the test _eventually_. ;-)

**Comment by bmonkman** [comment]
I think that is overkill since we are only toggling a single resource. If there were a bunch of things to do a module might make sense, but not just for one resource.

**Comment by davidcheung** [comment]
>I think that is overkill since we are only toggling a single resource. If there were a bunch of things to do a module might make sense, but not just for one resource.
theres some data source (the cf-secret itself)  and resources (k8s secret)

**Comment by bmonkman** [comment]
Actually yeah, if the issue here is that the secret that we are referring to in the data source will conditionally not exist, then excluding it by putting it in a submodule is probably the only way. Otherwise I wouldn't recommend a submodule for this amount of code.

**Comment by bmonkman** [comment]
I guess we could probably use `count` on the data source..

File: templates/kubernetes/terraform/modules/kubernetes/fileupload/main.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# File uploads
+
+data "aws_secretsmanager_secret" "cf_keypair" {
+  name = "${var.project}_cf_keypair"
```
Comment (bmonkman) [inline templates/kubernetes/terraform/modules/kubernetes/fileupload/main.tf:4]: You need to define variables for this module.

Comment (GrooveStomp) [reply templates/kubernetes/terraform/modules/kubernetes/fileupload/main.tf:4]: I was hoping it would "pass through". :blush:

Comment (bmonkman) [reply templates/kubernetes/terraform/modules/kubernetes/fileupload/main.tf:4]: No, all modules are equal. Whether it's defined in a subdirectory, somewhere else on disk, or an external repo, they all have the same definition and requirements. You can think of them as functions, where input variables are the function signature and output variables are the return values. The only thing that is ever passed implicitly is providers.

Comment (GrooveStomp) [reply templates/kubernetes/terraform/modules/kubernetes/fileupload/main.tf:4]: Ah, that's what I was thinking of. Thanks for explaining it. :+1:

**Review by bmonkman — APPROVED** [review]

File: templates/kubernetes/terraform/modules/kubernetes/fileupload/main.tf [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+locals {
+  cf_keypair_json = jsondecode(data.aws_secretsmanager_secret_version.cf_keypair.secret_string)
+}
+
+resource "kubernetes_secret" "cf_keypair" {
+  metadata {
+    name = "cf-keypair"
+    namespace = kubernetes_namespace.app_namespace.metadata[0].name
```
Comment (bmonkman) [inline templates/kubernetes/terraform/modules/kubernetes/fileupload/main.tf:18]: Namespace will need to be passed in

**Review by davidcheung — APPROVED** [review]

**Pull request merged**